File: src/main/java/ru/fullrest/mfr/plugins_configuration_utility/model/entity/Properties.java
package ru.fullrest.mfr.plugins_configuration_utility.model.entity;

import lombok.Data;

import javax.persistence.*;

/**
 * Created on 02.11.2018
 *
 * @author Alexey Plekhanov
 */
@Data
@Entity
@Table(name = "PROPERTY")
public class Properties {

    @Id
    @Column(name = "ID")
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;

    @Column(name = "KEY")
    @Enumerated(EnumType.STRING)
    private PropertyKey key;

    @Column(name = "VALUE")
    private String value;
}

File: desktop/application/src/main/kotlin/ru/fullrest/mfr/launcher/model/entity/Properties.kt
package ru.fullrest.mfr.launcher.model.entity

import javax.persistence.Column
import javax.persistence.Entity
import javax.persistence.EnumType
import javax.persistence.Enumerated
import javax.persistence.Table

@Entity
@Table(name = "property")
class Properties(

    @Column(name = "key")
    @Enumerated(EnumType.STRING)
    val key: Key,

    @Column(name = "value")
    var value: String? = null

) : BaseEntity() {
    enum class Key {
        SCHEMA, FIRST_START, SELECTED_BUILD, LAST_UPDATE_DATE
    }
}